src/classes/message/MessageRepository.js

```diff
@@ -1,18 +1,48 @@
 import axios from 'axios'
+import MessageMapper from "./MessageMapper";
 
 export default class MessageRepository {
 
-	async getAllMessages() {
-		const messages = (await axios.get('http://localhost:3000/messages', {
+	async getAllMessages(page) {
+		const messages = (await axios.get('http://localhost:3000/messages?_page='+page, {
+			headers: {
+				authorization: localStorage.getItem('token')
+			}
+		})).data;
+		return messages.map(MessageMapper.map);
+	}
+
+	async getMessagesByUser(id) {
+		const messages = (await  axios.get('http://localhost:3000/messages?author_id=' + id,  {
+			headers: {
+				authorization: localStorage.getItem('token')
+			}
+		})).data;
+		return messages.map(MessageMapper.map);
+	}
+
+	async getMessagesByHashtag(hashtag) {
+		return (await axios.get('http://localhost:3000/messages?chip_like=' + hashtag, {
 			headers: {
 				authorization: localStorage.getItem('token')
 			}
 		})).data
-		return messages
 	}
 
-	getMessagesByUser(id) {
-		return axios.get('http://localhost:3000/messages?author=' + id).then(response => response.data);
+	createMessage(message) {
```

Comment:
Why this is not async method? What user can do, if status 500?

Reply:
fixed